tensorstore/kvstore/s3/credentials/environment_credential_provider.cc

```diff
@@ -0,0 +1,64 @@
+// Copyright 2023 The TensorStore Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "tensorstore/kvstore/s3/credentials/environment_credential_provider.h"
+
+#include <optional>
+#include <string>
+
+#include "absl/log/absl_log.h"
+#include "absl/status/status.h"
+#include "absl/strings/str_cat.h"
+#include "absl/time/time.h"
+#include "tensorstore/internal/env.h"
+
+using ::tensorstore::internal::GetEnv;
+
+namespace tensorstore {
+namespace internal_kvstore_s3 {
+
+namespace {
+
+// AWS user identifier
+static constexpr char kEnvAwsAccessKeyId[] = "AWS_ACCESS_KEY_ID";
+// AWS user password
+static constexpr char kEnvAwsSecretAccessKey[] = "AWS_SECRET_ACCESS_KEY";
+// AWS session token
+static constexpr char kEnvAwsSessionToken[] = "AWS_SESSION_TOKEN";
+
+}  // namespace
+
+Result<AwsCredentials> EnvironmentCredentialProvider::GetCredentials() {
+  if (auto access_key = GetEnv(kEnvAwsAccessKeyId); access_key) {
```

Comment:
Nit. I prefer unnested / early return in general because I think that it flows better.

auto env = internal::GetEnv...
if (!env) {
  return absl::NotFoundError(...);
}
// other processing

Reply:
I've used the above style in the env cred provider and the file cred provider.